adapters/adapterstest/test_json.go

```diff
@@ -123,7 +123,9 @@ func runSpec(t *testing.T, filename string, spec *testSpec, bidder adapters.Bidd
 	diffErrorLists(t, fmt.Sprintf("%s: MakeBids", filename), bidsErrs, spec.MakeBidsErrors)
 
 	for i := 0; i < len(spec.BidResponses); i++ {
-		diffBidLists(t, filename, bidResponses[i].Bids, spec.BidResponses[i].Bids)
+		if bidResponses[i] != nil {
```

Comment:
I think the issue with this check is that if `bidResponses[i]` is `nil` but `spec.BidResponses[i]` is not `nil` then we are just skipping over that BidResponse and passing the test even though the test should be marked failed. How about changing the `diffBidLists` method to accept `[]*adapters.BidResponse` as an argument rather than `[]*adapters.TypedBid` and adding an assertion in that method to make sure that they are both either nil or not nil. What do you think?

I realize that this a gap in our test framework so If this is not too much for you to do and if you can then go ahead but if it is then lemme know and I can. put up a quick PR myself.

Reply:
Yeah, that makes sense. I'm in the middle of other work right now, but should be able to jump on it shortly.

Reply:
Perfect! Yeah, take your time. Thank you @muncha

Reply:
@mansinahar - I am looking at this, and I've tried a couple things along the lines of what you suggest above. I'm not quite there, but I think I get how it works. I should be updating the PR in an hour or two.

Reply:
But looking at this, I think maybe I should just return an empty adapters.NewBidderResponseWithBidsCapacity() instead of nil on 204.
```

// diffOrtbBids compares the actual Bid made by the adapter to the expectation from the JSON file.
func diffOrtbBids(t *testing.T, description string, actual *openrtb.Bid, expected json.RawMessage) {
	if actual == nil {
		t.Errorf("Bidders cannot return nil Bids. %s was nil.", description)
		return
	}
```